src/components/ComboBox/CarbonComboBox.jsx

```diff
@@ -0,0 +1,582 @@
+import React from 'react';
+import cx from 'classnames';
+import Downshift from 'downshift';
+import PropTypes from 'prop-types';
+import { settings } from 'carbon-components';
+import { Checkmark16, WarningFilled16 } from '@carbon/icons-react';
+import ListBox from 'carbon-components-react/es/components/ListBox';
+
+const keys = {
```

Comment:
I believe theres some keyCode constants already in the utils. Could you move this over there?

Reply:
This is from Carbon. With plans on pushing these changes back. There are some tweaks to the original Carbon component which is why we are temporarily carrying this component. I would rather not abstract these to another file only to have to delete them when this file is deleted.